torchqdynamics/odeint.py

```diff
@@ -0,0 +1,95 @@
+import warnings
+
+import torch
+
+# --------------------------------------------------------------------------------------
+#     Main odeint function
+# --------------------------------------------------------------------------------------
+
+
+def odeint(qsolver, y0, tsave, sensitivity='autograd', model_params=None):
+    # Check arguments
+    tsave = check_tsave(tsave)
+    if (model_params is not None) and (sensitivity in [None, 'autograd']):
+        warnings.warn('Argument `model_params` was supplied in `odeint` but not used.')
+
+    # Dispatch to appropriate odeint subroutine
+    if sensitivity is None:
+        return odeint_inplace(qsolver, y0, tsave)
```

Comment:
A good solution here might be to simply use
```python
if sensitivity is None:
    with torch.no_grad():
        return odeint_main(qsolver, y0, tsave)
```
This way torch does not store gradients. It still might be slower than an in-place solver, but it should give a performance boost.